### ISSUE TITLE
Error when preprocessing dataset: "Input type and weight type should be the same"

### ISSUE THREAD

**Issue by garland92** [filliptm/ComfyUI-FL-AceStep-Training#5]
Hi! I'm trying to train my first LoRA following the example workflow, but I'm getting an error when running the FL AceStep Preprocess Dataset node.

 _Log/Error Message_

> Input type (torch.cuda.HalfTensor) and weight type (torch.HalfTensor) should be the same

My Setup

- ComfyUI version: 0.13.0
- GPU: NVIDIA GeForce RTX 3060 Laptop GPU (6GB VRAM)
- PyTorch: 2.10.0+cu130
- CUDA: 13.0
- Model: ace_step_1.5_turbo_aio.safetensors https://huggingface.co/Comfy-Org/ace_step_1.5_ComfyUI_files/blob/main/checkpoints/ace_step_1.5_turbo_aio.safetensors

What I'm Doing

1. Following the example workflow: https://github.com/filliptm/ComfyUI-FL-AceStep-Training/blob/master/workflow/Example-WF.json
2. Using one test audio file (MP3 format)
3 Steps completed successfully:

- ✅ Load Checkpoint (model loads fine)
- ✅ Load LLM (0.6B version)
- ✅ Scan Audio Directory (finds my file)
- ✅ Auto-Label Samples (works perfectly - gets caption, BPM, key)
- ❌ Preprocess Dataset (fails with the error above)

What I've Tried

- Restarting ComfyUI
- Using different audio files
- Making sure all connections are correct in the workflow

Full Error Log
```
model weight dtype torch.bfloat16, manual cast: None
model_type FLOW
VAE load device: cuda:0, offload device: cpu, dtype: torch.float16
CLIP/text encoder model load device: cuda:0, offload device: cpu, current: cpu, dtype: torch.float16
model weight dtype torch.bfloat16, manual cast: None
model_type FLOW
[FL AceStep] Loading ACE-Step LLM: acestep-5Hz-lm-0.6B
Loading ACE-Step LLM: acestep-5Hz-lm-0.6B
[FL AceStep] LM model 'acestep-5Hz-lm-0.6B' is available
LM model 'acestep-5Hz-lm-0.6B' is available
[FL AceStep] Loading LLM from C:\StabilityMatrix\Data\Packages\ComfyUI\models\acestep\acestep-5Hz-lm-0.6B
Loading LLM from C:\StabilityMatrix\Data\Packages\ComfyUI\models\acestep\acestep-5Hz-lm-0.6B
[FL AceStep] Device: cuda, Backend: pt
Device: cuda, Backend: pt
Loading weights: 100%|██████████| 310/310 [00:02<00:00, 132.73it/s, Materializing param=model.norm.weight]
[FL AceStep] LLM 'acestep-5Hz-lm-0.6B' loaded successfully
LLM 'acestep-5Hz-lm-0.6B' loaded successfully
[FL AceStep] Scanning directory: C:\Users\Nature\Music\Miku_SongTest
Scanning directory: C:\Users\Nature\Music\Miku_SongTest
[FL AceStep] Found 1 audio files in C:\Users\Nature\Music\Miku_SongTest
Found 1 audio files in C:\Users\Nature\Music\Miku_SongTest
[FL AceStep] Found 1 audio files (tag: 'Future')
Found 1 audio files (tag: 'Future')
[FL AceStep] Starting auto-labeling...
Starting auto-labeling...
[FL AceStep] Moving audio tokenizer from cpu to cuda:0
Moving audio tokenizer from cpu to cuda:0
[FL AceStep] Encoding audio to codes for sample 0: 1-13. World is Mine.mp3
Encoding audio to codes for sample 0: 1-13. World is Mine.mp3
Requested to load AudioOobleckVAE
loaded completely;  321.70 MB loaded, full load: True
[FL AceStep] [ACEStep] audio_to_codes: 150 codes from C:\Users\Nature\Music\Miku_SongTest\1-13. World is Mine.mp3
[ACEStep] audio_to_codes: 150 codes from C:\Users\Nature\Music\Miku_SongTest\1-13. World is Mine.mp3
[FL AceStep] Generated 2994 chars of audio codes for sample 0, calling LLM...
Generated 2994 chars of audio codes for sample 0, calling LLM...
[FL AceStep] LLM understanding response length: 626
LLM understanding response length: 626
[FL AceStep] Sample 0 labeled: caption='A clean, arpeggiated electric guitar riff with a shimmering ...', bpm=150, key=E major
Sample 0 labeled: caption='A clean, arpeggiated electric guitar riff with a shimmering ...', bpm=150, key=E major
[FL AceStep] Labeled 1/1 samples
Labeled 1/1 samples
[FL AceStep] VAE loaded: dtype=torch.float16, device=cuda:0
VAE loaded: dtype=torch.float16, device=cuda:0
Requested to load ACE15TEModel_
loaded partially; 2140.13 MB usable, 0.00 MB loaded, 4672.68 MB offloaded, 2865.59 MB buffer reserved, lowvram patches: 0
[FL AceStep] CLIP/text encoder loaded
CLIP/text encoder loaded
[FL AceStep] Condition encoder: dtype=torch.bfloat16, device=cuda:0
Condition encoder: dtype=torch.bfloat16, device=cuda:0
[FL AceStep] Preprocessing 1 samples to ./output/acestep/datasets
Preprocessing 1 samples to ./output/acestep/datasets
[FL AceStep] Error processing sample sample_0000: Input type (torch.cuda.HalfTensor) and weight type (torch.HalfTensor) should be the same
Error processing sample sample_0000: Input type (torch.cuda.HalfTensor) and weight type (torch.HalfTensor) should be the same
[FL AceStep] Preprocessed 0/1 samples (1 errors)
Preprocessed 0/1 samples (1 errors)
Prompt executed in 25.13 seconds
```

**Comment by Vmars666** [comment]
same for me rtx5070t 12gvram
